react-delegationdashboard/src/pages/Ledger/AddressTable.tsx

```diff
@@ -0,0 +1,203 @@
+        hwWalletP
+          .getAccounts(startIndex, 5)
+          .then(accounts => {
+            setAccounts(accounts);
+            dispatch({ type: 'loading', loading: false });
+          })
+          .catch(err => {
+            setError('Check if Elrond app is open on Ledger');
+            console.error('error', err);
+            dispatch({ type: 'loading', loading: false });
+          });
+      })
+      .catch(err => {
+        if (err.statusCode in ledgerErrorCodes) {
+          setError((ledgerErrorCodes as any)[err.statusCode].message);
+        }
+        console.error('error', err);
+        dispatch({ type: 'loading', loading: false });
+      });
+  };
+
+  React.useEffect(fetchAccounts, [startIndex]);
+
+  const goToNext = () => {
+    setSelectedAddress('');
+    setStartIndex(current => current + 1);
+  };
+
+  const goToPrev = () => {
+    setSelectedAddress('');
+    setStartIndex(current => (current === 0 ? 0 : current - 1));
+  };
+
+  const setAddress = () => {
+    if (selectedIndex !== undefined) {
+      dispatch({
+        type: 'setLedgerAccount',
+        ledgerAccount: {
+          index: selectedIndex,
+          address: selectedAddress,
+        },
+      });
+      dispatch({ type: 'loading', loading: true });
+      const hwWalletProvider = new HWProvider(dapp.proxy, selectedIndex);
+      hwWalletProvider
+        .init()
+        .then((success: any) => {
+          if (!success) {
+            setError('could not initialise ledger app, make sure Elrond app is open');
+            dispatch({ type: 'loading', loading: false });
+            console.warn('could not initialise ledger app, make sure Elrond app is open');
+            return;
+          }
+
+          hwWalletProvider
+            .login()
+            .then(address => {
+              dispatch({ type: 'setProvider', provider: hwWalletProvider });
+              dispatch({ type: 'login', address });
+
+              dispatch({
+                type: 'ledgerLogin',
+                ledgerLogin: {
+                  index: selectedIndex,
+                  loginType: 'ledger',
+                },
+              });
+              history.push('/dashboard');
+            })
+            .catch((err: any) => {
+              if (err.statusCode in ledgerErrorCodes) {
+                setError((ledgerErrorCodes as any)[err.statusCode].message);
+                dispatch({
+                  type: 'setLedgerAccount',
+                  ledgerAccount: undefined,
+                });
+              }
+              dispatch({ type: 'loading', loading: false });
+              console.warn(err);
+            });
+        })
+        .catch((err: any) => {
+          if (err.statusCode in ledgerErrorCodes) {
+            setError((ledgerErrorCodes as any)[err.statusCode].message);
+          }
+          dispatch({ type: 'loading', loading: false });
+          console.warn('could not initialise ledger app, make sure Elrond app is open', err);
```

Comment:
also in all `catch` items in this file: maybe treat the error `Unknown error X` if the code is not inside known messages?